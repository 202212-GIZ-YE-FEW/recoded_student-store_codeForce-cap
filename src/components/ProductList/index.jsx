import { withTranslation } from "next-i18next"
import Image from "next/image"
import { AiOutlineHeart } from "react-icons/ai"

import { useGeneralCollection } from "@/utils/store"

function ProductList({ selectedFilter, t }) {
  const { data, error } = useGeneralCollection()
  // Remove duplicates from the products array
  if (error) {
    return <div>Error: {error.message}</div>
  }
  const uniqueProducts =
    data && Array.isArray(data)
      ? [...new Set(data.map((product) => JSON.stringify(product)))].map(
          (product) => JSON.parse(product)
        )
      : []

  // Filter the products based on the selected filter (if any)
  const categoryFilter = selectedFilter
    ? uniqueProducts.filter((product) =>
        product.category.includes(selectedFilter)
      )
    : uniqueProducts

  return (
    <div>
      <div
        className='grid xl:grid-cols-4 lg:grid-cols-4 md:grid-cols-2 sm:grid-cols-1 m-auto w-full'
        dir={t("language") === "ar" ? "rtl" : "ltr"}
      >
        {categoryFilter.map((product) => (
          <div
            key={product.uid}
            className='mx-3 mb-10 border rounded-lg cart-animation flex flex-col justify-between'
          >
            <div className='relative overflow-hidden'>
              <Image
                src={product.primaryImage.url}
                alt={product.productName}
                width={258}
                height={250}
                className='rounded-t-lg shadow-lg w-full bg-white'
              />
              <div className='absolute bottom-2 right-2 z-10'>
                <button className='flex items-center justify-center w-8 h-8 bg-white text-red-500 rounded-full shadow-md hover:text-red-500 transition-colors duration-300 ease-in-out'>
                  <AiOutlineHeart />
                </button>
              </div>
            </div>
            <div className='mx-3 text-center'>
              <div className='info flex justify-between my-4 mx-3'>
                <div className='text-left'>
                  <h2 className='font-semibold'>{product.productName}</h2>
                  <p className='font-extralight text-xs rtl:text-right'>
                    {product.category}
                  </p>
                </div>
                <div>
                  <h2 className='font-extrabold text-xl'>${product.price}</h2>
                  <p className='font-extralight text-xs'>{product.location}</p>
                </div>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  )
}

export default withTranslation("index")(ProductList)
