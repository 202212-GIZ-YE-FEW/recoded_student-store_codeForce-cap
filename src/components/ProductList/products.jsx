import { withTranslation } from "next-i18next"

const products = [
  {
    id: 1,
    name: "Moog Synthesizer",
    category: "Books",
    price: 10.0,
    made_city: "Istanbul / Turkey",
    order_date: "01/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 2,
    name: "Moog Synthesizer",
    category: "Furniture",
    price: 20.0,
    made_city: "Istanbul / Turkey",
    order_date: "03/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 3,
    name: "Moog Synthesizer",
    category: "Furniture",
    price: 30.0,
    made_city: "Istanbul / Turkey",
    order_date: "05/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 4,
    name: "Moog Synthesizer",
    category: "Electronics",
    price: 40.0,
    made_city: "Istanbul / Turkey",
    order_date: "07/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 5,
    name: "Moog Synthesizer",
    category: "Two-wheeler",
    price: 50.0,
    made_city: "Istanbul / Turkey",
    order_date: "09/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 6,
    name: "Moog Synthesizer",
    category: "Two-wheeler",
    price: 60.0,
    made_city: "Istanbul / Turkey",
    order_date: "11/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 7,
    name: "Moog Synthesizer",
    category: "Two-wheeler",
    price: 70.0,
    made_city: "Istanbul / Turkey",
    order_date: "13/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 8,
    name: "Moog Synthesizer",
    category: "Two-wheeler",
    price: 80.0,
    made_city: "Istanbul / Turkey",
    order_date: "15/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 9,
    name: "Moog Synthesizer",
    category: "Two-wheeler",
    price: 90.0,
    made_city: "Istanbul / Turkey",
    order_date: "17/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 10,
    name: "Moog Synthesizer",
    category: "Two-wheeler",
    price: 100.0,
    made_city: "Istanbul / Turkey",
    order_date: "19/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 11,
    name: "Moog Synthesizer",
    category: "Electronics",
    price: 200.0,
    made_city: "Istanbul / Turkey",
    order_date: "21/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 12,
    name: "Moog Synthesizer",
    category: "Electronics",
    price: 300.0,
    made_city: "Istanbul / Turkey",
    order_date: "23/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 13,
    name: "Moog Synthesizer",
    category: "Electronics",
    price: 500.0,
    made_city: "Istanbul / Turkey",
    order_date: "25/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 14,
    name: "Moog Synthesizer",
    category: "Electronics",
    price: 600.0,
    made_city: "Istanbul / Turkey",
    order_date: "27/04/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 15,
    name: "Moog Synthesizer",
    category: "Electronics",
    price: 700.0,
    made_city: "Istanbul / Turkey",
    order_date: "01/05/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
  {
    id: 16,
    name: "Moog Synthesizer",
    category: "Electronics",
    price: 800.0,
    made_city: "Istanbul / Turkey",
    order_date: "03/05/2023",
    listing_date: "25/2/2023",
    delivery_address: "X Apartment Y Street Z Province Istanbul/Turkey",
    images: "/images/productImg.png",
  },
]

export default withTranslation("index")(products)
