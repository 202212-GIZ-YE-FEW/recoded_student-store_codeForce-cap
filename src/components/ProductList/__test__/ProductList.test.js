import renderer from "react-test-renderer"

import ProductList from ".."

it("renders correctly", () => {
  const tree = renderer.create(<ProductList />).toJSON()
  expect(tree).toMatchSnapshot()
})
