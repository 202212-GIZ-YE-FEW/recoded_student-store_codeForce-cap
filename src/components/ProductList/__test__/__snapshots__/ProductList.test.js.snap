// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="grid xl:grid-cols-4 lg:grid-cols-4 md:grid-cols-2 sm:grid-cols-1 m-auto w-full"
    dir="ltr"
  >
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-one
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              10
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-two
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              20
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-two
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              30
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-three
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              40
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-four
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              50
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-four
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              60
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-four
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              70
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-four
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              80
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-four
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              90
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-four
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              100
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-three
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              200
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-three
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              300
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-three
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              500
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-three
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              600
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-three
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              700
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mx-3 mb-10 border rounded-lg cart-animation"
    >
      <img
        alt="name-product"
        className="rounded-t-lg shadow-lg w-full "
        data-nimg="1"
        decoding="async"
        height={211.41}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75"
        srcSet="/_next/image?url=%2Fimages%2FproductImg.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2FproductImg.png&w=640&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={258}
      />
      <div
        className="mx-3 text-center"
      >
        <div
          className="info flex justify-between my-4 mx-3"
        >
          <div
            className="text-left"
          >
            <h2
              className="font-semibold"
            >
              name-product
            </h2>
            <p
              className="font-extralight text-xs"
            >
              filter-three
            </p>
          </div>
          <div>
            <h2
              className="font-extrabold text-xl"
            >
              $
              800
            </h2>
            <p
              className="font-extralight text-xs"
            >
              made-city
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
